examples/default/src/screens/SurveysScreen.tsx

```diff
@@ -16,6 +45,46 @@ export const SurveysScreen: React.FC = () => {
         title="Show Custom Survey"
         onPress={() => Surveys.showSurvey('6ZaEI4nVdjg19r5uekS5nw')}
       />
+
+      <VerticalListTile title="Survey token actions">
+        <VStack>
+          <View style={styles.formContainer}>
+            <View style={styles.inputWrapper}>
+              <Input
+                placeholder="Survey token"
+                onChangeText={(key) => setSurveyToken(key)}
+                defaultValue={surveyToken}
+              />
+              {userAttributesFormError.surveyToken ? (
+                <Text style={styles.errorText}>{userAttributesFormError.surveyToken}</Text>
+              ) : null}
+            </View>
+          </View>
+
+          <Button
+            mt="4"
+            onPress={() => {
+              if (validateUserAttributeForm()) {
+                Surveys.showSurvey(surveyToken);
+              }
+            }}>
+            Show survey
+          </Button>
+
+          <Button
+            mt="4"
+            onPress={async () => {
+              if (validateUserAttributeForm()) {
+                const hasResponded = await Surveys.hasRespondedToSurvey(surveyToken);
+                toast.show({
+                  description: hasResponded ? 'YES' : 'NO',
+                });
+              }
```

Comment:
How about we refactor this to a method ?

Reply:
fixed